cpp/src/arrow/util/rle_encoding.h

```diff
@@ -414,6 +414,8 @@ static inline bool IndexInRange(int32_t idx, int32_t dictionary_length) {
 template <typename T>
 inline int RleDecoder::GetBatchWithDict(const T* dictionary, int32_t dictionary_length,
                                         T* values, int batch_size) {
+  using IndexType = int32_t;
```

Comment:
might be worth a comment here and below why IndexType is always static.  Would it be possible to add a unit test with a larger value that would have failed?

Reply:
You mean a larger type, or an index larger than 2**32?

Reply:
For Parquet use case, the max width of the index is 32 based on https://github.com/apache/parquet-format/blob/master/Encodings.md#dictionary-encoding-plain_dictionary--2-and-rle_dictionary--8

Reply:
I think I misunderstood the issue for the unit testing.  I guess this would have been caught if we had a big endian machine?  

I think adding the comment with the link that @kiszk provided would still make sense for the casual reader?

Reply:
Note that this was previously simply `int`. The change here is simply to make things clearer and also note the index size explicitly.